libmamba/src/solver/libsolv/database.cpp

```diff
@@ -109,7 +109,19 @@ namespace mamba::solver::libsolv
 
     void Database::set_logger(logger_type callback)
     {
-        ::pool_setdebuglevel(pool().raw(), std::numeric_limits<int>::max());  // All
+        // We must not use more than the penultimate level of verbosity of libsolv (which is 3) to
+        // avoid the most verbose messages (of type SOLV_DEBUG_RULE_CREATION | SOLV_DEBUG_WATCHES),
+        // which might spam the output and make mamba hang. See:
+        // https://github.com/openSUSE/libsolv/blob/27aa6a72c7db73d78aa711ae412231768e77c9e0/src/pool.c#L1623-L1637
+        // TODO: Make `level` configurable once the semantics and UX for verbosity are clarified.
+        // Currently, we use the behavior of `1.x` whose default value for the verbosity level was
+        // `0` in which case the logs of libsolv were not transferred. See:
+        // https://github.com/mamba-org/mamba/blob/4f269258b4237a342da3e9891045cdd51debb27c/libmamba/include/mamba/core/context.hpp#L88
+        // See: https://github.com/mamba-org/mamba/blob/1.x/libmamba/src/core/pool.cpp#L72
+        // Instead use something like:
+        // const int level = Context().output_params.verbosity - 1;
+        const int level = 0;
+        ::pool_setdebuglevel(pool().raw(), level);
```

Comment:
In 1.x we were doing:
`pool_setdebuglevel(pool().raw(), Context::instance().output_params.verbosity - 1);`
when `verbosity` >= 3
Otherwise `pool_setdebuglevel` wasn't called. That means that it was set to something in libsolv internals not to the default value of `verbosity` in `mamba`.

Reply:
[The default value of `libsolv`'s `pool->debugmask` is `SOLV_DEBUG_RESULT`](https://github.com/openSUSE/libsolv/blob/27aa6a72c7db73d78aa711ae412231768e77c9e0/src/pool.c#L87).

If we call it with `0` we only `&=SOLV_DEBUG_TO_STDERR` it. Not sure if it causes a problem in practice, but let's thus not call it for now at all.